e2e-tests/README.md

```diff
@@ -0,0 +1,143 @@
+
+## Building and testing the Operator with DockerHub
+
+There are scripts which build the image and run tests. Both building and testing
+needs some repository for the newly created docker images. If nothing is
+specified, scripts use Percona's experimental repository `perconalab/percona-server-mongodb-operator`, which
+obviously requires decent access rights to make a push.
+
+To specify your own repository for the Percona Server for MongoDB Operator docker image, you can use IMAGE environment variable:
+
+```
+export IMAGE=bob/my_repository_for_test_images:K8SPSMDB-372-fix-feature-X
+```
+
+Use the following script to build the image:
+
+```
+./e2e-tests/build
+```
+
+You can also build the image and run your cluster in one command:
+
+```
+./e2e-tests/build-and-run
+```
+
+Running all tests at once can be done with the following command:
+
+```
+./e2e-tests/run
+```
+
+Tests can be executed one-by-one, usingthe following scripts (their names  should be self-explanatory):
+
+
+```
+./e2e-tests/init-deploy/run
+./e2e-tests/arbiter/run
+./e2e-tests/limits/run
+./e2e-tests/scaling/run
+./e2e-tests/demand-backup/run
+./e2e-tests/scheduled-backup/run
+./e2e-tests/storage/run
+./e2e-tests/self-healing/run
+./e2e-tests/operator-self-healing/run
+....
+```
+
+## Building and testing the Operator without DockerHub
+
+The first thing you need to do is fixing image names, to use your custom Docker registry. For example, if your custom registry has the `172.30.162.173:5000` address, you can proceed with the following command:
+
+```
+export IMAGE=172.30.162.173:5000/namespace/repo:tag
+```
+
+Now you can use the following script to builds the image:
+
+```
+./e2e-tests/build
+```
+
+**Note:** there is no need to build an image if you are going to test some already-released version.
+
+Tests can be executed one-by-one, using the following scripts (their names should be self-explanatory):
+
+```
+./e2e-tests/init-deploy/run
+./e2e-tests/arbiter/run
+./e2e-tests/limits/run
+./e2e-tests/scaling/run
+./e2e-tests/demand-backup/run
+./e2e-tests/scheduled-backup/run
+./e2e-tests/storage/run
+./e2e-tests/self-healing/run
+./e2e-tests/operator-self-healing/run
+....
+```
+
+If the test failed, rerun it at least 3 times.
+
+## Using environment variables to customize the building and testing process
+
+### Re-declaring default image names
+
+You can use environment variables to re-declare all default images used for testing. The
+full list of variables is the following one:
+
+```
+IMAGE_MONGOD=${IMAGE_MONGOD:-"perconalab/percona-server-mongodb-operator:main-mongod4.4"}
```

Comment:
please see comment in pxco repo, I think it is better to use table with names of vars and default values because people can not understand what  `MAGE_MONGOD=${IMAGE_MONGOD:-"perconalab/percona-server-mongodb-operator:main-mongod4.4"}` means.

Reply:
turned into a list (similar to PXCO)